homeworks/01-bubble-sort/src/containers/BubbleSortMain/BubbleSortMain.tsx

```diff
@@ -0,0 +1,202 @@
+                                />
+                            )
+                        }
+                    </div>
+                </main>
+                <footer>
+                    <p>{this.state.status}</p>
+                    <div>
+                        <label htmlFor='order'>Desc Order</label>
+                        <input id='order'
+                            type="checkbox"
+                            checked={!this.state.isAscOrder}
+                            disabled={this.state.status === SortingStatus.IN_PROGRESS || this.state.status === SortingStatus.ON_PAUSE}
+                            onChange={(event) => this.changeSortOrder(event)}
+                        />
+                    </div>
+
+                    <div>
+                        <label htmlFor='interval'>Sorting Interval</label>
+                        <input id='interval'
+                            value={this.state.sortIterval}
+                            type="range"
+                            min="100" max="1000" step="100"
+                            disabled={this.state.status === SortingStatus.IN_PROGRESS}
+                            onChange={(event) => this.changeSortInterval(event)}
+                        />
+                    </div>
+
+                    <div>
+                        <label htmlFor='array'>Array Length</label>
+                        <input id='array'
+                            type="range"
+                            value={this.state.length}
+                            min="5" max="100" step="5"
+                            disabled={this.state.status === SortingStatus.IN_PROGRESS || this.state.status === SortingStatus.ON_PAUSE}
+                            onChange={(event) => this.changeSortingArray(event)}
+                        />
+                    </div>
+
+                    <button onClick={() => this.start()}
+                        disabled={this.state.status !== SortingStatus.ON_START}
+                    >
+                        Start
+                    </button>
+                    <button onClick={() => this.reset()}
+                        disabled={this.state.status === SortingStatus.IN_PROGRESS}
+                    >
+                        Reset
+                    </button>
+                    <button onClick={() => this.iterate()}
+                        disabled={this.state.status === SortingStatus.SOLVED}
+                    >
+                        Iterate
+                    </button>
+                    <button onClick={() => this.pause()}
+                        disabled={this.state.status === SortingStatus.SOLVED || this.state.status === SortingStatus.ON_START}
+                    >
+                        {this.state.status === SortingStatus.ON_PAUSE ? 'Resume' : 'Pause'}
+                    </button>
+                </footer>
+            </div>
+        );
+    }
+
+}
+
+export default BubbleSortMain;
```

Comment:
Also, you could split the main component and its logic by components because this file increases and become really big.